h2o-core/src/main/java/water/Job.java

```diff
@@ -480,4 +481,42 @@ private void update_from_remote( ) {
       }
   }
   @Override public Class<KeyV3.JobKeyV3> makeSchema() { return KeyV3.JobKeyV3.class; }
+
+  @Override
+  @SuppressWarnings("unchecked")
+  protected void copyOver(Job src) {
+    super.copyOver(src);
+    _barrier = src._barrier;
+  }
```

Comment:
is it not used by the icing/weaving/serialization logic? what would it mean to serialize a transient attribute like a Barrier? Did you try this in multinode?

Reply:
although if you explicitly overrode this method, there must be a reason… which I don't get immediately

Reply:
gotcha, in the `tryGetDoneJob` below, you need to retrieve the job by key... and then you need the barrier.
But does it make sense to keep the barrier transient then?

Reply:
the barrier is not h2o-serializable, it is imho not conceptually serializable either - it needs to be transient

Reply:
I agree that conceptually, it's weird to consider the barrier as serializable, but I guess I misunderstand the semantic of `copyOver`:
from Weaver.java:
```java
    // DTasks need to be able to copy all their (non transient) fields from one
    // DTask instance over another, to match the MRTask API.
    if( iced_cc.subclassOf(_dtask) ) {
      String cpbody_impl =
        make_body(icer_cc, iced_cc, iced_clazz, "copyOver", null, null,
                  "  protected void copyOver(water.Freezable fdst, water.Freezable fsrc) {\n",
```
I understood from the comments above that `copyOver` was supposed to copy only non-transient fields, but the few time it's being overridden, it's to copy transient ones, so I'm definitely missing something. Hard to find out exactly how it is used...

Reply:
I need to check it is really necessary - it might not be, thanks for the comment

Reply:
yes, I think that is a misususe (= doesn't do anything) Job is not descendant from DTask so it doesn't apply here